common.sh

```diff
@@ -41,17 +41,10 @@ cd "$base_dir" || {
   exit 1
 }
 
-go_workspace=''
-for p in ${GOPATH//:/ }; do
-  if [[ $PWD/ = $p/* ]]; then
-    go_workspace=$p
-  fi
-done
-
-if [ -z $go_workspace ]; then
-  echo 'Current directory is not in $GOPATH' >&2
-  exit 1
-fi
+# create a temporary gopath, and clean it up (with some sanity checks to ensure
+# we don't do anything terrible)
+tmp_gopath=$(mktemp --tmpdir -d kubebuilder-test-gopath-XXXXXX)
```

Comment:
This [fails on OSX](https://travis-ci.org/kubernetes-sigs/kubebuilder/jobs/544391172) with `mktemp: illegal option -- -`

I suggest that this be changed to `mktemp -d "${TMPDIR:-/tmp}/kubebuilder-test-gopath-XXXXXX"` as it should work regardless of platform.